File: searching_algs/binary_search/binary_search.js
let binarySearch = (arr, l, r, key) => {
    if (r >= l) {
        let mid = l + Math.floor((r - l) / 2);
        if (arr[mid] === key) {
            return mid;
        }
        if (arr[mid] > key) {
            return binarySearch(arr, l, mid - 1, key);
        }
        return binarySearch(arr, mid + 1, r, key);
    }
    return -1;
}

let main = () => {
    let arr = [1, 2, 3, 4, 5, 6, 7];
    let len = arr.length - 1;
    let key = 5;
    let test1 = binarySearch(arr, 0, len, key);
    let test2 = binarySearch(arr, 0, len, 0);

    if(test1 == 4) {
        console.log("Element found test 1 \t\t passed");
    } else {
        console.log("Element not found test 1 \t\t failed");
    }

    if(test2 == -1) {
        console.log("Element not found test 2 \t passed");
    } else {
        console.log("Element not found test 2 \t failed");
    }
    return 0;
};

main();

File: searching_algs/binary_search/binary_search.c
#include <stdio.h>

int binarySearch(int arr[], int l, int r, int key) {
    if(r>=l){
      int m = l + (r-l)/2;
      if(arr[m] == key){
        return m;
      }
      if(arr[m]>key){
        return binarySearch(arr, l, m-1, key);
      }
      return binarySearch(arr, m+1, r, key);
    }
    return -1;
}

int main(void) {
    int arr[] = {1, 2, 3, 4, 5, 6, 7};
    int len = sizeof(arr) / sizeof(arr[0]) - 1;
    int key = 5;

    int test1 = binarySearch(arr, 0, len, key);
    int test2 = binarySearch(arr, 0, len, 0);

    if(test1 == 4) {
        printf("Element found test 1 \t\t passed \n");
    } else {
        printf("Element not found test 1 \t\t failed \n");
    }

    if(test2 == -1) {
        printf("Element not found test 2 \t passed \n");
    } else {
        printf("Element not found test 2 \t failed \n");
    }
}